src/tiled/mainwindow.cpp

```diff
@@ -813,115 +905,11 @@ void MainWindow::export_()
-        suffix.prepend(QLatin1String("*."));
-
-        for (MapFormat *format : helper.formats()) {
-            if (format->nameFilter().contains(suffix, Qt::CaseInsensitive)) {
-                if (chosenFormat) {
-                    QMessageBox::warning(this, tr("Non-unique file extension"),
-                                         tr("Non-unique file extension.\n"
-                                            "Please select specific format."));
-                    return exportAs();
-                } else {
-                    chosenFormat = format;
-                }
-            }
-        }
-    }
-
-    if (!chosenFormat) {
-        QMessageBox::critical(this, tr("Unknown File Format"),
-                              tr("The given filename does not have any known "
-                                 "file extension."));
-        return;
-    }
-
-    // Check if writer will overwrite existing files here because some writers
-    // could save to multiple files at the same time. For example CSV saves
-    // each layer into a separate file.
-    QStringList outputFiles = chosenFormat->outputFiles(mapDocument->map(),
-                                                        fileName);
-    if (outputFiles.size() > 0) {
-        // Check if any output file already exists
-        QString message =
-                tr("Some export files already exist:") + QLatin1String("\n\n");
-
-        bool overwriteHappens = false;
-
-        for (const QString &outputFile : outputFiles) {
-            if (QFile::exists(outputFile)) {
-                overwriteHappens = true;
-                message += outputFile + QLatin1Char('\n');
-            }
-        }
-        message += QLatin1Char('\n') + tr("Do you want to replace them?");
-
-        // If overwrite happens, warn the user and get confirmation before exporting
-        if (overwriteHappens) {
-            const QMessageBox::StandardButton reply = QMessageBox::warning(
-                this,
-                tr("Overwrite Files"),
-                message,
-                QMessageBox::Yes | QMessageBox::No,
-                QMessageBox::No);
-
-            if (reply != QMessageBox::Yes)
-                return;
-        }
+    if (auto mapDocument = qobject_cast<MapDocument*>(mDocument)) {
+        exportMapAs(mapDocument);
     }
-
-    pref->setLastPath(Preferences::ExportedFile, QFileInfo(fileName).path());
-    mSettings.setValue(QLatin1String("lastUsedExportFilter"), selectedFilter);
-
-    if (!chosenFormat->write(mapDocument->map(), fileName)) {
-        QMessageBox::critical(this, tr("Error Exporting Map"),
-                              chosenFormat->errorString());
-    } else {
-        // Remember export parameters, so subsequent exports can be done faster
-        mapDocument->setLastExportFileName(fileName);
-        mapDocument->setExportFormat(chosenFormat);
+    else if (auto tilesetDocument = qobject_cast<TilesetDocument*>(mDocument)) {
```

Comment:
Coding style: `else` on the same line as the `}`.

Reply:
Done